env.d.ts

```diff
@@ -14,6 +14,15 @@ type ScriptFailure = {
 type ScriptResponse<T = object> = ScriptSuccess<T> | ScriptFailure
 type ErrorScripts = Record<string, () => ScriptFailure>
 
+type Scriptor = Pick<Function, "name" | "call">;
```

Comment:
a scriptor is not a `Function`. it happens to also have a `.name` and `.call()` method but it's not a `Function` and its type shouldn't be derived from `Function`. I like the `Scriptor` type in the bellow comment think it's on the right track